idanalysis/idsdata.py

```diff
@@ -1191,3 +1191,90 @@ def _get_kmap_config_name(phase, de):
         de_str = Tools.get_phase_str(de)
         fname += "_de_{}".format(de_str)
         return fname
+
+
+class WLSData(Tools):
+    """WLS data access and manipulation class."""
+
+    PARAMS = _PARAMS()
+    # PARAMS.B_PEAK = 1.0  # [T]
+    # PARAMS.PERIOD_LEN = 180  # [mm]
+    # PARAMS.ID_LEN = 2.654  # [m]
+    # PARAMS.NR_PERIODS = 13
+    PARAMS.KPARAMETER_NAME = "current"
+    PARAMS.ID_FAMNAME = "WLS"
+    PARAMS.SUBSECS = ["ID14SB"]
+
+    PARAMS.KICKMAPS_DATA_PATH = REPOS_PATH + "wls/kickmaps/"
+    PARAMS.FIELDMAPS_DATA_PATH = (
+        DATA_REPOS_PATH
+        + "lnls-ima/si-wls/model-09/simulation/magnetic/updated-BHcurves/"
+        + "magnet-results/field-maps-two-currents/"
+    )
+    PARAMS.FOLDER_BASE_OUTPUT = REPOS_PATH + "wls/results/data/"
+
+    FIELMAPS_CONFIGS = {
+        'I=1A': 'SWLS_I=1A_X=-10_10mm_Y=-1.75_1.75mm_Z=-1000_1000mm.txt',
+        'I=10A': 'SWLS_I=10A_X=-10_10mm_Y=-1.75_1.75mm_Z=-1000_1000mm.txt',
+        'I=20A': 'SWLS_I=20A_X=-10_10mm_Y=-1.75_1.75mm_Z=-1000_1000mm.txt',
+        'I=40A': 'SWLS_I=40A_X=-10_10mm_Y=-1.75_1.75mm_Z=-1000_1000mm.txt',
+        'I=60A': 'SWLS_I=60A_X=-10_10mm_Y=-1.75_1.75mm_Z=-1000_1000mm.txt',
+        'I=80A': 'SWLS_I=80A_X=-10_10mm_Y=-1.75_1.75mm_Z=-1000_1000mm.txt',
+        'I=100A': 'SWLS_I=100A_X=-10_10mm_Y=-1.75_1.75mm_Z=-1000_1000mm.txt',
+        'I=120A': 'SWLS_I=120A_X=-10_10mm_Y=-1.75_1.75mm_Z=-1000_1000mm.txt',
+        'I=140A': 'SWLS_I=140A_X=-10_10mm_Y=-1.75_1.75mm_Z=-1000_1000mm.txt',
+        'I=160A': 'SWLS_I=160A_X=-10_10mm_Y=-1.75_1.75mm_Z=-1000_1000mm.txt',
+        'I=200A': 'SWLS_I=200A_X=-10_10mm_Y=-1.75_1.75mm_Z=-1000_1000mm.txt',
+        'I=227A': 'SWLS_I=227A_X=-10_10mm_Y=-1.75_1.75mm_Z=-1000_1000mm.txt',
+        'I=250A': 'SWLS_I=250A_X=-10_10mm_Y=-1.75_1.75mm_Z=-1000_1000mm.txt',
+    }
+
+    CURR_DICT = {
+        (1): 'I=1A',
+        (10): 'I=10A',
+        (20): 'I=20A',
+        (40): 'I=40A',
+        (60): 'I=60A',
+        (80): 'I=80A',
+        (100): 'I=100A',
+        (120): 'I=120A',
+        (140): 'I=140A',
+        (160): 'I=160A',
+        (200): 'I=200A',
+        (227): 'I=227A',
+        (250): 'I=250A',
+    }
```

Comment:
why do you have both dictionaries? why not define only one from the numeric value of the current to the fieldmap config?